### PR TITLE
butterflypack: add v3.2.0, strumpack: add v8.0.0

### PR DESCRIPTION
**Pull request by balay**

### CODE REVIEW

**Review by liuyangzhuan — APPROVED** [review]

**Review by wdconinc — COMMENTED** [review]
Did you verify that the `%fj` patches still apply correctly to the new `butterflypack` release? If not, please add a conflict for `%fj` for this newer version and possibly alert the person who added the patches by tagging them. As maintainer you may see benefit in applying patches for all compilers (whenever possible), even if the patches were introduced for a specific compiler.

**Review by wdconinc — COMMENTED** [review]
Verified both checksums.

**Comment by balay** [comment]
> Did you verify that the %fj patches still apply correctly to the new butterflypack release? 

 I see this was added in 05ffbbc0befa26d520612b7e801c26d7e8bda28f via #17514 

I'll suggest `%fj` conflicts should be addressed in a separate PR - perhaps @ketsubouchi can check on it

If its needed here -  I can push:
```diff
diff --git a/var/spack/repos/builtin/packages/butterflypack/package.py b/var/spack/repos/builtin/packages/butterflypack/package.py
index 9385175d6e..dba0573767 100644
--- a/var/spack/repos/builtin/packages/butterflypack/package.py
+++ b/var/spack/repos/builtin/packages/butterflypack/package.py
@@ -62,9 +62,9 @@ class Butterflypack(CMakePackage):
     # https://github.com/spack/spack/issues/31818
     patch("qopenmp-for-oneapi.patch", when="@2.1.1 %oneapi")
 
-    patch("longline.patch", when="%fj")
-    patch("fjfortran.patch", when="%fj")
-    patch("isnan.patch", when="%fj")
+    patch("longline.patch", when="%fj @1.2.0")
+    patch("fjfortran.patch", when="%fj @1.2.0")
+    patch("isnan.patch", when="%fj @1.2.0")
 
     def cmake_args(self):
         spec = self.spec
```
as this is the version of butterflypack when this patch was added.

File: var/spack/repos/builtin/packages/butterflypack/package.py
Comment (wdconinc) [inline var/spack/repos/builtin/packages/butterflypack/package.py]: To avoid restricting the path to `fj@1.2.0`, don't we need this?
```suggestion
    patch("longline.patch", when="@1.2.0 %fj")
    patch("fjfortran.patch", when="@1.2.0 %fj")
    patch("isnan.patch", when="@1.2.0 %fj")
```

**Review by wdconinc — APPROVED** [review]
I'll approve, but would be good to get @pghysels to approve as well before merging.

**Review by pghysels — APPROVED** [review]

**Pull request merged**
